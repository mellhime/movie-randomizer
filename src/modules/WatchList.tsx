import { FC } from "react";
import { useFetch } from "@hooks/useFetch";
import { DataTable } from "primereact/datatable";
import { Column } from "primereact/column";

const WatchList: FC = () => {
  const [watchList] = useFetch("watchList");

  const handleClick = (id: number) => {
    console.log(id);
    // todo call api remove from watch list
  };

  const editButton = (movie) => {
    return (
      <button onClick={() => handleClick(movie.id)}>
        Remove from watch list
      </button>
    );
  };

  return (
    <>
      <DataTable value={watchList} tableStyle={{ minWidth: "50rem" }}>
        <Column field="title" header="Title" />
        <Column body={editButton} header="Edit button" className="w-10rem" />
      </DataTable>
    </>
  );
};

export { WatchList };
