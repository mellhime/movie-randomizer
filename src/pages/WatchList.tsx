import { FC } from "react";
import { useFetch } from "../hooks/useFetch";

const WatchList: FC = () => {
  const [watchList] = useFetch("watchList");

  const handleClick = (id: number) => {
    console.log(id);
    // todo call api remove from watch list
  };

  const tableBody = watchList.map((movie) => (
    <tr key={movie.id}>
      <td>{movie.title}</td>
      <td>
        <button onClick={() => handleClick(movie.id)}>
          Remove from watch list
        </button>
      </td>
    </tr>
  ));

  return (
    <>
      <table>
        <thead>
          <tr>
            <th>Название фильма</th>
            <th></th>
          </tr>
        </thead>
        <tbody>{tableBody}</tbody>
      </table>
    </>
  );
};

export { WatchList };
