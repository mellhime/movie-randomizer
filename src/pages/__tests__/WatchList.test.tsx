import { render, renderHook } from "@testing-library/react";
import { WatchList } from "../WatchList";
import { useFetch } from "../../hooks/useFetch";

describe("WatchList component", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<WatchList />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("useFetch hook", () => {
  it("gets data", () => {
    const { result } = renderHook(() => useFetch("watchList"));

    expect(result.current[0][0]).toHaveProperty("title");
  });
});
