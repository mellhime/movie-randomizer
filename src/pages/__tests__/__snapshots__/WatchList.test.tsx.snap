// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WatchList component matches snapshot 1`] = `
<DocumentFragment>
  <table>
    <thead>
      <tr>
        <th>
          Название фильма
        </th>
        <th />
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Leon
        </td>
        <td>
          <button>
            Remove from watch list
          </button>
        </td>
      </tr>
      <tr>
        <td>
          Neon Genesis Evangelion
        </td>
        <td>
          <button>
            Remove from watch list
          </button>
        </td>
      </tr>
      <tr>
        <td>
          2012
        </td>
        <td>
          <button>
            Remove from watch list
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
